gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/kubernetes/KubernetesSyncService.java

```diff
@@ -73,18 +78,46 @@ protected void doStart() throws Exception {
     private void startWatch() {
         logger.info("Kubernetes synchronization started at {}", Instant.now().toString());
         this.disposable =
-            watch()
+            watchConfigMaps()
                 .flatMapCompletable(this::handleConfigMapEvent)
                 .doOnError(throwable -> logger.error("An error occurred during configmaps refresh. Restarting watch.", throwable))
                 .retry()
                 .subscribe();
     }
 
-    private Flowable<Event<ConfigMap>> watch() {
+    private Flowable<Event<ConfigMap>> watchConfigMaps() {
+        if (namespace != null) {
+            if ("ALL".equals(namespace)) {
+                return watchConfigMaps(null);
+            } else if (namespace.contains(",")) { // multiple comma separated namespace "ns1,ns2,ns3"
+                String[] namespaces = this.namespace.split(",");
+                // Just create one Websocket connection and filter the results
+                return watchConfigMaps(null)
+                    .filter(
+                        configMapEvent -> {
+                            for (String ns : namespaces) {
```

Comment:
(node sure) What do you think of trim the string? so that it works too `ns1, ns2, ns3`. 

but wouldn't a yaml array work too?

Reply:
Label is a good way to do that but it's more risky, labels are cluster scoped. If we have two GKO on the same cluster (for 2 differents team). They can make mistakes and read the same label! It's safer to use NS... (not my preferred choice but safer)